src/socket.ts

```diff
@@ -345,7 +344,9 @@ export class FluentSocket extends EventEmitter {
     ) {
       if (this.state === SocketState.DISCONNECTING) {
         // Try again once the socket has fully closed
-        await awaitNextTick();
+        await new Promise(resolve =>
+          this.once(FluentSocketEvent.CLOSE, resolve)
+        );
```

Comment:
huh, I wonder why I didn't just do this in the first place 🤷 I guess I read somewhere in the node docs that it will be disconnected by the next tick? not sure where I got that from.